lib/apiservers/engine/backends/container.go

```diff
@@ -1032,11 +1053,18 @@ func (c *Container) ContainerTop(name string, psArgs string) (*types.ContainerPr
 
 // Containers returns the list of containers to show given the user's filtering.
 func (c *Container) Containers(config *types.ContainerListOptions) ([]*types.Container, error) {
+	defer trace.End(trace.Begin(fmt.Sprintf("ListOptions %#v", config)))
```

Comment:
Yeah, I was thinking the same thing.  We may not need an entire research spike.  Maybe we can keep a list of items.  One of the items we can do is to split the code up into functional files like Docker.  That should help with multiple hands in the cookie jar, but that brings up the question of benefits since we don't know when the containerd work will be done.